### PR TITLE
fix proc signature in list_ops

### PR DESCRIPTION
**Pull request by antonio-petrillo**
I thinkg the proc signature for `ls_foldl` and `ls_foldr` in [list-ops](https://exercism.org/tracks/odin/exercises/list-ops) is wrong.  
The type of the accumulator function is `proc(acc, element: T) -> $U`, the `initial_value` is `T`, with these types after a *first accumulation* we get a value of type `U` that cannot be passed into the accumulation function.  
There is also an edge case when the empty list is being passed in input, we can just return the zero value for the type `U` which is not the `initial_value` passed as input.
Considering for example `ls_foldl`:  
```odin 
ls_foldl :: proc(l: []$T, initial_value: T, fn: proc(acc, element: T) -> $U) -> U ---
```
Should be
```odin 
ls_foldl :: proc(l: []$T, initial_value: $U, fn: proc(acc: U, element: T) -> U) -> U ---
```
At the moment the tests passes because they involve always the same type as input and as initial value, I've added a simple test for this edge case.

### CODE REVIEW

File: exercises/practice/list-ops/list_ops_test.odin [hunk 1]
```diff
@@ -163,6 +163,22 @@ test_return_a_list_of_elements_whose_values_equal_the_list_value_transformed_by_
...
+	{ 	// test for foldl
+		result := ls_foldl(list, f64(0), sum_f64_and_int)
+		testing.expect_value(t, result, 55.0)
+	}
+	{ 	// test for foldr
+		result := ls_foldr(list, f64(0), sum_f64_and_int)
+		testing.expect_value(t, result, 55.0)
+	}
+}
+
```
Comment (rmonnet) [inline exercises/practice/list-ops/list_ops_test.odin]: This is a good test to add but it may be better to have one test for each procedure (`foldl` and `foldr`), this allows the student to see right away why the test fail and follows the spirit of TDD.

```suggestion
@(test)
/// description = foldl (reduces) the given list to the type of accumulator
test_foldl_reduces_the_given_list_to_the_type_of_accumulator :: proc(t: ^testing.T) {
	sum_f64_and_int :: proc(acc: f64, el: int) -> f64 { return acc + f64(el) }

	list := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
	result := ls_foldl(list, f64(0), sum_f64_and_int)
	testing.expect_value(t, result, 55.0)
}

@(test)
/// description = foldr (reduces) the given list to the type of accumulator
test_foldr_reduces_the_given_list_to_the_type_of_accumulator :: proc(t: ^testing.T) {
	sum_f64_and_int :: proc(acc: f64, el: int) -> f64 { return acc + f64(el) }

	list := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
	result := ls_foldr(list, f64(0), sum_f64_and_int)
	testing.expect_value(t, result, 55.0)
}
```
